packages/lesswrong/components/posts/PostsPage/CollapsedFootnotes.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useState, useRef, useEffect } from "react";
+import { registerComponent } from "../../../lib/vulcan-lib";
+import { useLocation } from "../../../lib/routeUtil";
+import Collapse from "@material-ui/core/Collapse";
+import classNames from "classnames";
+import { useOnSearchHotkey } from "../../common/withGlobalKeydown";
+
+export const EXPAND_FOOTNOTES_EVENT = "expand-footnotes";
+
+export const locationHashIsFootnote = (hash: string) =>
+  hash.startsWith("#fn") && !hash.startsWith("#fnref");
+
+const styles = (theme: ThemeType) => ({
+  collapse: {
+    marginTop: "-1.5em",
+  },
+  fullyExpanded: {
+    overflow: "visible",
+  },
+  buttonWrapper: {
+    marginTop: "1em",
+  },
+  button: {
+    fontFamily: theme.palette.fonts.sansSerifStack,
+    fontSize: 16,
+    fontWeight: 600,
+    color: theme.palette.primary.main,
+    cursor: "pointer",
+    "&:hover": {
+      color: theme.palette.primary.light,
+    },
+  },
+});
+
+const htmlToElement = <T extends HTMLElement>(html: string): T | null => {
+  const div = document.createElement("div");
+  div.innerHTML = html.trim();
+  return div.firstChild as T | null;
+}
+
+const extractChildren = <T extends HTMLElement>(
+  elem: T,
+  selector: string,
+): string => {
+  const children = Array.from(elem.querySelectorAll(selector));
+  return children.map((child) => child.outerHTML).join(" ");
+}
+
+// Because this component is only created client-side by ContentItemBody, this
+// doesn't need to be (and definitely isn't) SSR-safe.
+const splitFootnotes = (html: string, previewCount: number) => {
+  const elem = htmlToElement<HTMLDivElement>(html);
+  return elem
+    ? {
+      preview: extractChildren(elem, `li:nth-child(-n + ${previewCount})[id]`),
+      rest: extractChildren(elem, `li:nth-child(n + ${previewCount + 1})[id]`),
+      totalCount: elem.querySelectorAll("li").length,
+    }
+    : {preview: "", rest: "", totalCount: 0};
+}
+
+const CollapsedFootnotes = ({
```

Comment:
1. This isn't working for me, when I load a page with a footnote hash in the url I can briefly see the footnote but then it disappears
2. I would quite like some way of seeing all the footnotes at once, sometimes the footnotes are the bulk of the post

Is this here so that the bottom of the post isn't too busy? I would be ok with merging without it. Usually there aren't that many footnotes, and people will scroll to the comments anyway so they are likely to see the recommendations

Reply:
1. I couldn't reproduce the issue. What post did you try it on - I'm wondering if maybe there are some posts that use differently formatted markup than what I was expecting (for instance, I'd expect [http://localhost:3000/s/isENJuPdB3fhjWYHd/p/7JxsXYDuqnKMqa6Eq](http://localhost:3000/s/isENJuPdB3fhjWYHd/p/7JxsXYDuqnKMqa6Eq) to work correctly).
2. There should be a "show all footnotes" button - maybe it's not there for reasons related to (1)?

Reply:
[This](https://beta.effectivealtruism.org/posts/oCeENfefZkAQhEh5b/feature-announcement-rich-text-editor-footnotes#fncys7a23lnfa) is the post I was looking at, which definitely does have the footnotes on [prod](https://forum.effectivealtruism.org/posts/oCeENfefZkAQhEh5b/feature-announcement-rich-text-editor-footnotes#fncys7a23lnfa). I wonder if it's do to with not having enough to make the "show all" appear?